### PR TITLE
Graph NG: fix toggling queries and extract Graph component from graph3 panel

### PR DESCRIPTION
**Pull request by dprokop**
- Fixed issue with graph3 panel breaking when showing/hiding queries .
- Splitted out the GraphNG component from Graph3/GraphPanel for reusability (cc @zoltanbedi please take a look, when this is merged you can remove GraphPanel usage from exemplars branch and replace it with GraphNG component with a limited API). This component takes data frames and is reposnsible for scales, series and axis configuration while still actepting Graph NG plugins as children.
- Added some tests coverage for most important behaviours of graph ng (data update, plugins API and uPlot reinitialization)
- Fixed grid color in dark theme
- Fixed annotations rerender bug on query change

Ref https://github.com/grafana/grafana/issues/27848

### CODE REVIEW

File: packages/grafana-ui/src/components/GraphNG/GraphNG.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+const defaultFormatter = (v: any) => (v == null ? '-' : v.toFixed(1));
+
+const TIME_FIELD_NAME = 'Time';
+
+interface GraphNGProps extends Omit<PlotProps, 'data'> {
+  data: DataFrame[];
+}
+
+export const GraphNG: React.FC<GraphNGProps> = ({ data, children, ...plotProps }) => {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/GraphNG/GraphNG.tsx:69]: This component will replace the current Graph component in grafana/ui when we replace the current graph in Explore

File: packages/grafana-ui/src/components/GraphNG/GraphNG.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    seriesIdx++;
+  }
+
+  return (
+    <UPlotChart data={alignedData} {...plotProps}>
+      {scales}
+      {axes}
+      {geometries}
+      {children}
```
Comment (dprokop) [inline packages/grafana-ui/src/components/GraphNG/GraphNG.tsx:186]: `children` property allows graph plugins(context menu, toolitp etc) composition

File: packages/grafana-ui/src/components/uPlot/Plot.tsx [hunk 3]
```diff
@@ -59,22 +53,38 @@ export const UPlotChart: React.FC<PlotProps> = props => {
...
-    if (!currentConfig) {
+  useLayoutEffect(() => {
+    // Make sure everything is ready before proceeding
+    if (!currentConfig || !plotData.current) {
+      return;
+    }
+
+    // Do nothing if there is data vs series config mismatch. This may happen when the data was updated and made this
+    // effect fire before the config update triggered the effect.
+    if (currentConfig.series.length !== plotData.current.length) {
```
Comment (dprokop) [inline packages/grafana-ui/src/components/uPlot/Plot.tsx:69]: @mckn this is the actual bug fix... Not super happy about this flow yet, think would be good if we could sit down at some point and think about improving the uPlot init/data update logic

File: packages/grafana-ui/src/components/uPlot/hooks.test.ts [hunk 4]
```diff
@@ -88,7 +86,7 @@ describe('usePlotConfig', () => {
           stroke: '#00ff00',
         });
       });
-      await waitForNextUpdate();
+      // await waitForNextUpdate();
```
Comment (dprokop) [inline packages/grafana-ui/src/components/uPlot/hooks.test.ts:89]:
```suggestion
```

File: packages/grafana-ui/src/components/uPlot/hooks.test.ts [hunk 5]
```diff
@@ -132,7 +130,7 @@ describe('usePlotConfig', () => {
 
         removeSeries();
       });
-      await waitForNextUpdate();
+      // await waitForNextUpdate();
```
Comment (dprokop) [inline packages/grafana-ui/src/components/uPlot/hooks.test.ts:133]:
```suggestion
```

File: packages/grafana-ui/src/components/uPlot/hooks.test.ts [hunk 6]
```diff
@@ -165,15 +163,15 @@ describe('usePlotConfig', () => {
...
+      const { result } = renderHook(() => usePlotConfig(0, 0, 'browser'));
       const addAxis = result.current.addAxis;
 
       act(() => {
         addAxis({
           side: 1,
         });
       });
-      await waitForNextUpdate();
+      // await waitForNextUpdate();
```
Comment (dprokop) [inline packages/grafana-ui/src/components/uPlot/hooks.test.ts:174]:
```suggestion
```

**Review by mckn — APPROVED** [review]
I think this looks good and works as expected! Great work with this @dprokop!

As you already mentioned that we probably need to see if we can simply parts of the structure a bit but let's do that in future PRs.

**Pull request merged**